autopilot/followpoint.h

```diff
@@ -33,40 +36,44 @@ class FollowPoint : public QObject
 
     bool isActive();
 
-    void setPosTypeUsed(const PosType &posTypeUsed);
-
 signals:
     void deactivateEmergencyBrake();
     void activateEmergencyBrake();
 
 public slots:
-    void updateFollowPointInVehicleFrame(const PosPoint &point);
+    void pointToFollowInVehicleFrame(const PosPoint &point);
+    void pointToFollowInEnuFrame(const PosPoint &point);
+
 
 private:
-    const unsigned mFollowPointTimeout_ms = 1000;
+    unsigned mFollowPointTimeout_ms = 1000;
+    bool mFollowPointTimedOut = true;
     unsigned mUpdateStatePeriod_ms = 50;
     QTimer mFollowPointHeartbeatTimer;
     QTimer mUpdateStateTimer;
 
-    PosPoint mCurrentGoal;
-    FollowPointSTMstates mStmState = FollowPointSTMstates::NONE;
-
-    PosPoint mCurrentFollowPointInVehicleFrame; // independent of positioning
+    PosPoint mCurrentPointToFollowInEnuFrame;
+    double mDistanceToPointIn2D;
+    QLineF mLineFromVehicleToPoint;
```

Comment:
Would suggest to put the dynamic state that is used to follow into a single struct to clearly separate STM stuff from configuration parameters etc., e.g.: https://github.com/RISE-Dependable-Transport-Systems/WayWise/blob/8533e0096ca6fffe0b201ad6e619cf94ae1180f0/autopilot/purepursuitwaypointfollower.h#L22